### PR TITLE
#12704 - Cases can be created for Contacts, Event participants and Tr…

### LINKED ISSUE #12704: Cases can be created for Contacts, Event participants and Travel entries when the CASE SURVEILLANCE is disabled
### Bug Description

When the CASE SURVEILLANCE is disabled, cases can still be created for contacts, event participants and travel entries via the case create buttons that are still displayed on the respective pages.

Similar to #12643 but as the flows were not mentioned in the ticket to begin with I created the separate issue.

### Reproduction Steps

Contacts:
1. In the DB set the CASE_SURVEILLANCE to 'false';
2. Log in the web app with an Admin+NatUser;
3. Open a contact that has the status 'Active contact';
4. Click to make the status 'Confirmed Contact' and 'Save';
5. Then click on 'Create a case for this contact person';
6. Fill in the mandatory information and 'Save'.

Event Participants:
1. In the DB set the CASE_SURVEILLANCE to 'false';
2. Log in the web app with an Admin+NatUser;
3. Open an existing event and navigate to the 'Event participants' list;
4. For one of the participants that does not have a case, click on 'Create' from the column 'Case ID';
5. Fill in the mandatory information and 'Save'.

Travel Entries:
1. In the DB set the CASE_SURVEILLANCE to 'false';
2. Log in the web app with an Admin+NatUser;
3. Open an existing travel entry;
4. Click on the button 'Create case';
5. Fill in the mandatory information and 'Save'.

Actual behavior: The new cases are saved and the case page is displayed.

### Expected Behavior

The buttons to create cases should not be displayed when the CASE SURVEILLANCE featureconfiguration is disabled.

### Affected Area(s)

Web app

### SORMAS Version

1.91.0; 1.90.0

### Android version/Browser

Chrome 

### User Role(s)

Admin+NatUser

### Server URL

local machine 

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by sergiupacurariu**
…avel entries when the CASE SURVEILLANCE is disabled

Fixes #12704

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12744

File: sormas-ui/src/main/java/de/symeda/sormas/ui/travelentry/TravelEntryDataView.java [hunk 1]
```diff
@@ -65,7 +65,9 @@ protected void initView(String params) {
 		container.addComponent(layout);
 
 		UserProvider currentUser = UserProvider.getCurrent();
-		boolean caseButtonVisible = currentUser != null && currentUser.hasUserRight(UserRight.CASE_CREATE);
+		boolean caseButtonVisible = currentUser != null
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/travelentry/TravelEntryDataView.java:68]: you could use `UiUtil.isPermitted`

File: sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsGrid.java [hunk 2]
```diff
@@ -130,14 +131,16 @@ public EventParticipantsGrid(EventParticipantCriteria criteria) {
...
 
-		addItemClickListener(new ShowDetailsListener<>(CASE_ID, false, e -> {
-			if (e.getCaseUuid() != null) {
-				ControllerProvider.getCaseController().navigateToCase(e.getCaseUuid());
-			} else if (e.getInJurisdiction()) {
-				EventParticipantDto eventParticipant = FacadeProvider.getEventParticipantFacade().getEventParticipantByUuid(e.getUuid());
-				ControllerProvider.getCaseController().createFromEventParticipant(eventParticipant);
-			}
-		}));
+		if (FacadeProvider.getFeatureConfigurationFacade().isFeatureEnabled(FeatureType.CASE_SURVEILANCE)) {
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsGrid.java:134]: you should check also the CASE_VIEW right, please use `UiUtil.isPermitted`

File: sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsGrid.java [hunk 3]
```diff
@@ -80,7 +80,8 @@ public EventParticipantsGrid(EventParticipantCriteria criteria) {
 		caseIdColumn.setSortProperty(EventParticipantIndexDto.CASE_UUID);
 		caseIdColumn.setRenderer(new CaseUuidRenderer(uuid -> {
 			// '!=' check is ok because the converter returns the constant when no case creation is allowed
-			return NO_CASE_CREATE != uuid;
+
+			return NO_CASE_CREATE != uuid && FacadeProvider.getFeatureConfigurationFacade().isFeatureEnabled(FeatureType.CASE_SURVEILANCE);
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsGrid.java:84]: Should check CASE_CREATE right too, please use `UiUtil.isPermitted`

File: sormas-ui/src/main/java/de/symeda/sormas/ui/contact/ContactDataView.java [hunk 4]
```diff
@@ -99,7 +99,9 @@ protected void initView(String params) {
 		editComponent = ControllerProvider.getContactController()
 			.getContactDataEditComponent(getContactRef().getUuid(), ViewMode.NORMAL, contactDto.isPseudonymized());
 
-		addCreateFromCaseButtonLogic();
+		if (FacadeProvider.getFeatureConfigurationFacade().isFeatureEnabled(FeatureType.CASE_SURVEILANCE)) {
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/contact/ContactDataView.java:102]: Should check CASE_CREATE right too, please use `UiUtil.isPermitted`

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-ui/src/main/java/de/symeda/sormas/ui/travelentry/TravelEntryDataView.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/travelentry/TravelEntryDataView.java:68]: changed

File: sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsGrid.java (hunk 2 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsGrid.java:134]: changed

File: sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsGrid.java (hunk 3 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventParticipantsGrid.java:84]: changed

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12744

File: sormas-ui/src/main/java/de/symeda/sormas/ui/contact/ContactDataView.java (hunk 4 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/contact/ContactDataView.java]: updated

**Review by leventegal-she — APPROVED** [review]

**Pull request merged**